tests/unit/test_main.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import sys
+
+from unittest.mock import patch, Mock
+
+from auditwheel.main import main
+
+
+def test_non_linux_platform(monkeypatch, capsys):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "darwin")
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 1
+    captured = capsys.readouterr()
+    assert captured.out == "Error: This tool only supports Linux\n"
+
+
+@patch("auditwheel.main.argparse.ArgumentParser")
+@patch("auditwheel.main.logging.basicConfig")
+def test_verbose_logging_non_verbose(basic_cfg_mock, parser_mock, monkeypatch):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "linux")
+    args = Mock()
+    args.verbose = 0
+    args.func.return_value = 0
+    parser_mock.return_value.parse_args.return_value = args
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 0
+    basic_cfg_mock.assert_called_once_with(level=logging.INFO)
+
+
+@patch("auditwheel.main.argparse.ArgumentParser")
+@patch("auditwheel.main.logging.basicConfig")
+def test_verbose_logging_verbose(basic_cfg_mock, parser_mock, monkeypatch):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "linux")
+    args = Mock()
+    args.verbose = 1
+    args.func.return_value = 0
+    parser_mock.return_value.parse_args.return_value = args
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 0
+    basic_cfg_mock.assert_called_once_with(level=logging.DEBUG)
```

Comment:
Again, how are these going to fail? If we change the argument parsing or the logging setup, we'd just have to change the tests to match. I find it hard to imagine any situation where these tests would fail and we'd change the `main()` function instead of the test.

Reply:
While I agree that the code tested here is unlikely to change without the test needing to change, if you simply want to make sure that the `-v` flag works correctly, how else do you implement the test for it? 

If we say that we don't test verbose mode because it would couple the test with the implementation, I would still opt to have the test, just so that it's validated. I think the only other option is to check that text the logger outputs includes `DEBUG`, but that would have to make assumptions on other parts of the system, which is worse in my opinion.

Reply:
I probably just wouldn't write a test for it. It works. There's no need to change it, and if we do switch to a new logging framework or something, the test will fail regardless of whether there's a bug.

If I really, really needed to test something like this, I'd try to find some operation which is guaranteed to produce debug level logging and, as you mention, verify that some of that is shown. That's ultimately what we care about. But I agree with you that this is rather an ugly test, and I probably wouldn't bother with it.

Reply:
Same as @takluyver, I wouldn't bother with these 2 tests.